modules/programs/quickshell.nix

```diff
@@ -0,0 +1,90 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.quickshell;
+in
+{
+  meta.maintainers = [ lib.hm.maintainers.justdeeevin ];
+
+  options.programs.quickshell = {
+    enable = lib.mkEnableOption "quickshell, a flexbile QtQuick-based desktop shell toolkit.";
+    package = lib.mkPackageOption pkgs "quickshell" { nullable = true; };
+    configs = lib.mkOption {
+      type = lib.types.attrsOf lib.types.path;
```

Comment:
Might be nice to support `lib.types.lines` in the future with `lib.hm.types.sourceFileOrLines`. I don't know if it is setup to support this usecase of a dynamic location, atm.